### PR TITLE
[Quant][fx][resubmit] Remove dequant-quant around getitem

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83172

Summary: https://github.com/pytorch/pytorch/issues/82480 saw
unnecessary dequant-quant pairs around the getitem op, which led
to significant slowdowns. This commit simply removes this pair in
the lowering step, since getitem already handles quantized inputs.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_getitem_no_dequant_quant

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Tasks: https://github.com/pytorch/pytorch/issues/82480

Differential Revision: [D38657297](https://our.internmc.facebook.com/intern/diff/D38657297)

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @ezyang @SherlockNoMad @soumith @EikanWang @wenzhe-nrv

### CODE REVIEW

**Comment by jerryzh168** [comment]
was the internal test error resolved?

**Comment by andrewor14** [comment]
I'm still trying to reproduce it. Making a PR first so I can import

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38657297).

**Comment by jerryzh168** [comment]
I think getitem support is explicitly skipped right now: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/prepare.py#L402-L408, maybe we can enable this instead?

**Comment by andrewor14** [comment]
> I think getitem support is explicitly skipped right now: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/prepare.py#L402-L408, maybe we can enable this instead?

I remember this broke some tests that were difficult to fix in the original PR. I think it had to do with `getitem` being used for dict inputs. Since this is a resubmit maybe we should revisit this in the future?

**Comment by andrewor14** [comment]
> I think getitem support is explicitly skipped right now: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/prepare.py#L402-L408, maybe we can enable this instead?

As discussed offline, we cannot enable this without breaking the dict/list of tensors use case. I added a detailed explanation in the comments for context.

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38657297).

**Review by jerryzh168 — APPROVED** [review]
looks great, thanks!

File: torch/ao/quantization/fx/prepare.py [hunk 1]
```diff
@@ -1377,7 +1377,35 @@ def insert_observers_for_model(
                         node_name_to_target_dtype_info, node_name_to_qconfig,
                         model, modules, graph)
 
+        # Second pass: Look for getitem nodes and make the input and output observers the same.
         #
+        # Context: There are two use cases of getitem:
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/prepare.py:1382]: should we move this to the doc for `_make_getitem_share_input_output_observers`?

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38657297).

**Pull request closed without merging**